labcontrol/db/process.py

```diff
@@ -3819,3 +3826,407 @@ def _generate_amplicon_prep_information(self):
+
+                id = d['stub_lot_id']
+                d['stub_lot_id'] = reagent[id]['external_lot_id']
+
+                # We have two robot IDs. Not sure which one is rightfully the
+                # 'extraction robot', but the example value is both strings
+                # separated by an underscore. Tentatively using this combined
+                # value.
+                id = d['gepmotion_robot_id']
+                epm_robot = equipment[id]['external_id']
+                id = d['kingfisher_robot_id']
+                kf_robot = equipment[id]['external_id']
+                d['extraction_robot'] = '%s_%s' % (epm_robot, kf_robot)
+
+                # Note extraction_kit_id references (as in foreign-key)
+                # reagent_composition(reagent_composition_id).
+                id = d['extraction_kit_id']
+                d['extraction_kit_lot'] = reagent[id]['external_lot_id']
+
+                id = d['epmotion_tool_id']
+                d['epmotion_tool_name'] = equipment[id]['external_id']
+
+                # for now, platform is hard-coded to 'Illumina'
+                # will need to change once Nanopore is supported by LC
+                # and we have a column to record one or the other.
+                d['platform'] = 'Illumina'
+
+                # these key/value pairs are tentatively hard-coded for now.
+                d['sequencing_method'] = 'sequencing by synthesis'
+                d['run_center'] = 'UCSDMI'
+                d['library_construction_protocol'] = 'KL KHP'
+
+                # EXPERIMENT_DESIGN_DESCRIPTION as with Amplicon, will remain
+                # empty when NULL.
+
+                # Replicating logic from Amplicon pre-processing
+                # TODO: refactor to a shared method
+                d['orig_name2'] = d['orig_name']
+
+                # Note is_control field replaces the line directly below
+                if not d['is_control']:
+                    # strip the prepended study id from orig_name2, but only
+                    # if this is an 'experimental sample' row, and not a
+                    # 'control' row. (captured here w/orig_name2 and study_id
+                    # not equaling None. This also prevents interference w/the
+                    # population of the DataFrame index below, using the
+                    # existing list comprehension.
+                    d['orig_name2'] = re.sub("^%s\." % d['study_id'],
+                                             '',
+                                             d['orig_name2'])
+
+            return results
+
+    def _generate_metagenomics_prep_information(self):
+        """Generates prep information for Metagenomics workflows
+
+        An internal method used to implement the generation of prep information
+        files for Metagenomics workflows. This method is called by
+        generate_prep_information() only. This method is not currently
```

Comment:
It seems like there is an implementation below :)

Reply:
👍